repo_policy_compliance/blueprint.py

```diff
@@ -124,6 +126,26 @@ def one_time_token() -> str:
     return token
 
 
+@repo_policy_compliance.route(POLICY_ENDPOINT, methods=["POST"])
+@auth.login_required(role=CHARM_ROLE)
+def policy_endpoint() -> Response:
+    """Generate a one time token for a runner.
+
+    Returns:
+        Either that the policy was updated or an error if the policy is invalid.
+    """
+    # Need to use this to update variable in outside scope, part of what is required to not require
+    # a database
+    global policy_document  # pylint: disable=global-statement
```

Comment:
Why not using a session ?

Reply:
Not sure a session would work - it conceptually is tied to a user and the policy needs to apply globally. I'll switch this to use a temporary file that exists for the life of the flask app

Reply:
Well then I feel like there's something wrong in the process.
Can you describe the workflow the GH Runner will use that will include this endpoint ?
My feeling is that, if it's global, why isn't the policy part of the application configuration, why does it needs an endpoint ?

Reply:
The runners don't directly use this endpoint, they still use check-run. The behaviour of check-run is modified by the policy that applies. The policy that applies is decided by the charm who would be the user of this endpoint. The configuration would be set through configuration of the charm.

A configuration file is an alternative. The policy endpoint can be thought of as a management API to influence the behaviour of the application. The advantage of an endpoint is that it is easier to know how to interact with it, i.e., just send a request. It is also easier to set the configuration programmatically rather than translate configuration to file which is then read by the application again. Discourse, for example, also has a configuration API. With a configuration file there is always the question of where it is located, was it found by the application etc. Environment variables would also be an alternative.

The end state of this is that the charm uses pebble notices in which case this kind of management API is no longer needed as the charm will know which policies are configured based on the charm configuration.